cargo-bolero tests fail to build from crates.io

Trying to update Nix `cargo-bolero` package (which [obtains the source](https://github.com/NixOS/nixpkgs/blob/e46ce3db0494637b54faa166d437fa6e6d3e2ae3/pkgs/development/tools/rust/cargo-bolero/default.nix#L7-L10) from crates.io), I get the following test failures:

```
   Compiling cargo-bolero v0.11.1 (/build/cargo-bolero-0.11.1.tar.gz)
error[E0433]: failed to resolve: use of undeclared crate or module `bolero`
 --> tests/fuzz_harnessed/fuzz_target.rs:3:5
  |
3 |     bolero::check!().for_each(|_| {});
  |     ^^^^^^ use of undeclared crate or module `bolero`

error[E0432]: unresolved import `bolero`
 --> tests/fuzz_generator/fuzz_target.rs:1:5
  |
1 | use bolero::check;
  |     ^^^^^^ use of undeclared crate or module `bolero`

error[E0432]: unresolved import `bolero`
 --> tests/fuzz_bytes/fuzz_target.rs:1:5
  |
1 | use bolero::check;
  |     ^^^^^^ use of undeclared crate or module `bolero`

```

Looking at the contents of the [crate archive](https://static.crates.io/crates/cargo-bolero/cargo-bolero-0.11.1.crate), it is because `Cargo.toml` no longer contains `bolero` dependency:

```toml
# THIS FILE IS AUTOMATICALLY GENERATED BY CARGO
#
# When uploading crates to the registry Cargo will automatically
# "normalize" Cargo.toml files for maximal compatibility
# with all versions of Cargo and also rewrite `path` dependencies
# to registry (e.g., crates.io) dependencies.
#
# If you are reading this file be aware that the original Cargo.toml
# will likely look very different (and much more reasonable).
# See Cargo.toml.orig for the original contents.

# [omitted…]

[dev-dependencies]

[features]
afl = ["bolero-afl"]
default = [
    "afl",
    "libfuzzer",
    "kani",
]
honggfuzz = ["bolero-honggfuzz"]
kani = []
libfuzzer = []
```

It is present in `Cargo.toml.orig`, which is probably why it works on CI.

--- a/bin/cargo-bolero/Cargo.toml
+++ b/bin/cargo-bolero/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "cargo-bolero"
-version = "0.11.1"
+version = "0.11.2"
 authors = ["Cameron Bytheway <bytheway.cameron@gmail.com>"]
 description = "cargo command for running bolero fuzz tests"
 homepage = "https://github.com/camshaft/bolero"
@@ -34,7 +34,7 @@ tar = "0.4"
 tempfile = "3"
 
 [dev-dependencies]
-bolero = { path = "../../lib/bolero" }
+bolero = { version = "0.11", path = "../../lib/bolero" }
 
 [[test]]
 name = "fuzz_bytes"

--- a/lib/bolero-engine/Cargo.toml
+++ b/lib/bolero-engine/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "bolero-engine"
-version = "0.11.0"
+version = "0.11.2"
 authors = ["Cameron Bytheway <bytheway.cameron@gmail.com>"]
 description = "fuzz and property testing framework"
 homepage = "https://github.com/camshaft/bolero"
@@ -27,7 +27,7 @@ rand_xoshiro = { version = "0.6", optional = true }
 backtrace = { version = "0.3", default-features = false, features = ["std"] }
 
 [dev-dependencies]
-bolero-generator = { path = "../bolero-generator", features = ["std"] }
+bolero-generator = { version = "0.11", path = "../bolero-generator", features = ["std"] }
 rand = "0.8"
 rand_xoshiro = "0.6"
 

